launcher/downloader/__init__.py

```diff
@@ -3,6 +3,8 @@
 from .base import Base
 
 from urllib.parse import urlparse
+from tenacity import retry, TryAgain, stop_after_attempt
+from pathlib import Path
 
```

Comment:
You are missing `from requests.exceptions import ConnectionError` which is used at L30

Reply:
Line added